app/src/pages/inside/logsPage/logItemInfo/logItemInfoTabs/stackTrace/stackTrace.jsx

```diff
@@ -0,0 +1,183 @@
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import Parser from 'html-react-parser';
+import classNames from 'classnames/bind';
+import { injectIntl, intlShape, defineMessages } from 'react-intl';
+import {
+  logItemsSelector,
+  activeLogIdSelector,
+  logPaginationSelector,
+  NAMESPACE,
+  DEFAULT_LOG_LEVEL,
+} from 'controllers/log';
+import { pagePropertiesSelector } from 'controllers/pages';
+import ArrowIcon from 'common/img/arrow-down-inline.svg';
+import { activeProjectSelector } from 'controllers/user';
+import { fetch } from 'common/utils/fetch';
+import { URLS } from 'common/urls';
+import { SpinningPreloader } from 'components/preloaders/spinningPreloader';
+import { NoItemMessage } from 'components/main/noItemMessage';
+import styles from './stackTrace.scss';
+
+const cx = classNames.bind(styles);
+
+const messages = defineMessages({
+  messageRefCaption: {
+    id: 'StackTrace.messageRefCaption',
+    defaultMessage: 'Go to stack trace in log message',
+  },
+  noStackTrace: {
+    id: 'StackTrace.noStackTrace',
+    defaultMessage: 'No stack trace to display',
+  },
+});
+
+@connect((state) => ({
+  logItemId: activeLogIdSelector(state),
+  projectId: activeProjectSelector(state),
+  pagination: logPaginationSelector(state),
+  pageProperties: pagePropertiesSelector(state, NAMESPACE),
+  logItems: logItemsSelector(state),
+}))
+@injectIntl
+export class StackTrace extends Component {
+  static propTypes = {
+    intl: intlShape.isRequired,
+    projectId: PropTypes.string.isRequired,
+    logItemId: PropTypes.string.isRequired,
+    onChangePage: PropTypes.func.isRequired,
+    pagination: PropTypes.object.isRequired,
+    pageProperties: PropTypes.object.isRequired,
+    logItems: PropTypes.array.isRequired,
+  };
+
+  static checkIfStackTraceItemOnThisPage = (logItems, stackTraceItem) =>
+    !!logItems.find((item) => item.id === stackTraceItem.id);
+
+  static getDerivedStateFromProps(props, state) {
+    return state.activePage !== props.pagination.number
+      ? {
+          activePage: props.pagination.number,
+          isStackTraceItemOnThisPage: StackTrace.checkIfStackTraceItemOnThisPage(
+            props.logItems,
+            state.stackTraceItem,
+          ),
+        }
+      : null;
+  }
+
+  constructor(props) {
+    super(props);
+    this.state = {
```

Comment:
You can set initial state without constructor.
```
state = {
  ...
}
```

Reply:
Fixed